libs/cardano-ledger-test/src/Test/Cardano/Ledger/Generic/AggPropTests.hs

```diff
@@ -105,3 +122,61 @@ forAllChainTrace p@(Allegra _) n propf =
   property $ propf <$> genTrace p n (def {blocksizeMax = 4, slotDelta = (6, 12)}) (initStableFields p)
 forAllChainTrace p@(Shelley _) n propf =
   property $ propf <$> genTrace p n (def {blocksizeMax = 4, slotDelta = (6, 12)}) (initStableFields p)
+
+-- ===========================================================
+
+-- | Check that the sum of dsDeposits and the psDepoits are equal to the utsosDeposits
+depositInvariant ::
+  SourceSignalTarget (MOCKCHAIN era) ->
+  Property
+depositInvariant SourceSignalTarget {source = mockChainSt} =
+  let LedgerState {lsUTxOState = utxost, lsDPState = DPState dstate pstate} = (esLState . nesEs . mcsNes) mockChainSt
+      allDeposits = utxosDeposited utxost
+      sumCoin m = Map.foldl' (<+>) (Coin 0) m
```

Comment:
This is almost exact monomorphic definition of [`sumVal`](https://github.com/input-output-hk/cardano-ledger/blob/b613b13139c9be9ba19c57cd2ab73a9b64446dd4/libs/cardano-ledger-core/src/Cardano/Ledger/Val.hs#L95)
```suggestion
      sumCoin m = Map.foldl' (<+>) (Coin 0) m
```

Reply:
Yes but then again, these things are going to end up CompactCoin, so the changes will be easier to make if this is explicit. Some thing like

Map.foldl' addCompact (CompactCoin 0) m